[REFACTOR] 가독성을 위한 리팩터링

jd99iam opened the pull request:
## #116 

### 💡 PR 타입(하나 이상의 PR 타입을 선택해주세요)
* [x] 리팩터링

### 📲 반영 브랜치
ex) refactor/116-project -> 1-develop

### ❔ 변경 사항
for문을 최대한 줄이고 가독성을 위해 stream을 사용했습니다
반복문을 돌며 dto에 더하는 코드를 수정해서 stream을 바로 배열로 만들도록 했습니다
메서드를 최대한  추출해서 가독성을 좋게 했습니다

### 🔍 테스트 결과

### 📰 참고 Reference

### ⚒️ 기타 사항
SolutionPostRequest의 toProblem에서 ProblemRepository를 인자로 받아서 저장하는데, dto에서 problemRepository를 인자로 받아서 처리하는 것에 대해 의견이 궁금합니다

jd99iam committed:
fix: BadgeConstant 씨앗 삭제

Related to : #116

jd99iam committed:
refactor: ReviewService 리팩터링

1. for문 사용하지 않도록 수정
2. Collect 사용하도록 수정

Related to : #116

jd99iam committed:
refactor: ProblemService 리팩터링

1. for문 최소화
2. 가독성 좋게 수정

Related to : #116

jd99iam committed:
refactor: SolutionService 리팩터링

1. getSolutionDetail 가독성 좋게 리팩터링

Related to : #116

jd99iam committed:
refactor: SolutionDetailResponse 리팩터링

1. stream 사용하도록 리팩터링
2. 메서드 추출로 from 메서드 간소화

Related to : #116

jd99iam committed:
refactor: MemberProfileResponse 리팩터링

1. 상수 사용하도록 변경
2. 메서드 추출

Related to : #116

jd99iam committed:
refactor: SolutionReviewsResponse 리팩터링

1. lineNumber가 있어서 for문 사용
2. 메서드 추출

Related to : #116

jd99iam committed:
refactor: 작명 변경

Related to : #116

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/member/response/MemberProfileResponse.java:
```diff
@@ -44,27 +45,42 @@ public static MemberProfileResponse from(Member member, List<Badge> badges, List
-        if (solutions.size()<3){
+        if (checkLastScroll(solutions)){
             isLast = true;
         } else {
```
`else` 예약어를 사용하지 않는 방법은 없을까요 ?

jd99iam replied:
> `else` 예약어를 사용하지 않는 방법은 없을까요 ?

여기서는 불가피해 보입니다

jd99iam replied:
> `else` 예약어를 사용하지 않는 방법은 없을까요 ?

커서값으로 마지막인지 알 수 있으니 isLast를 사용하지 않도록 하고 else문을 뺐습니다

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/review/response/SolutionReviewsResponse.java:
```diff
@@ -24,14 +24,20 @@ public class SolutionReviewsResponse {
         // 먼저 최종 응답 dto에 각 라인을 넣는다
-        for (int lineNumber = 0; lineNumber <lines.length; lineNumber++) {
+        for (int lineNumber = 0; lineNumber < lines.length; lineNumber++) {
             SolutionLine solutionLine = SolutionLine.builder()
```
solutionLine에 대한 정적 팩토리 메서드는 없나요 ?

jd99iam replied:
> solutionLine에 대한 정적 팩토리 메서드는 없나요 ?

수정했습니다

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/solution/response/SolutionDetailResponse.java:
```diff
@@ -22,47 +22,45 @@ public class SolutionDetailResponse {
-                        comment.getStatus(),
-                        new ArrayList<>()
-                )));
+                .forEach(comment-> commentResponseList.add(new SolutionDetailComment(comment.getMember().getPhoto(), comment.getId(), comment.getMember().getNickname(), comment.getContent(), comment.getStatus(), new ArrayList<>())));
```
SolutionDetailComment에 from 메서드를 만들면 SoltionDetailComment.from(comment)로 해결 될 것 같아요

jd99iam replied:
> SolutionDetailComment에 from 메서드를 만들면 SoltionDetailComment.from(comment)로 해결 될 것 같아요

수정했습니다

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/solution/response/SolutionDetailResponse.java:
```diff
@@ -22,47 +22,45 @@ public class SolutionDetailResponse {
+    private static List<SolutionDetailComment> makeCommentsResponse(List<Comment> comments) {
+        List<SolutionDetailComment> commentResponseList = new ArrayList<>();
         // 먼저 부모가 없는 댓글들을 전부 더한다
         comments.stream().filter(comment -> comment.getParentId()==null)
```
`comments.stream().filter(comment -> comment.getParentId()==null).map(SolutionDetailComment::from).collect(Collectors.toList())` 와 같은 방식으로 하는 건 어떤가요 ?

jd99iam replied:
> `comments.stream().filter(comment -> comment.getParentId()==null).map(SolutionDetailComment::from).collect(Collectors.toList())` 와 같은 방식으로 하는 건 어떤가요 ?

수정했습니다

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/service/ProblemService.java:
```diff
@@ -44,33 +47,41 @@ private static List<ProblemListProblem> makeProblemList(List<Problem> problems)
-            if (tag != null) {
-                problemListProblem.getTags().add(keySet.get(tagCount));
-            }
+    private static void addTagToProblemListProblem(ProblemListProblem problemListProblem, List<Solution> solutions) {
```
static 메서드인 이유가 있나요 ?

jd99iam replied:
> static 메서드인 이유가 있나요 ?

static 뺐습니다

jd99iam committed:
Merge branch '1-develop' of https://github.com/rlfrkdms1/Prog-rangers into refactor/116-project

jd99iam committed:
refactor: isLast를 사용하지 않도록 변경

1. isLast를 사용하기보다 cursor값이 -1이면 마지막 스크롤인것으로 판단하도록 변경
2. 변경에 다른 else문 삭제

Related to : #116

jd99iam committed:
refactor: SolutionLine에 정적 팩터리 메서드 추가

Related to : #116

jd99iam committed:
refactor: SolutionDetailComment에 from 추가

Related to : #116

jd99iam committed:
refactor: SolutionDetailComment에 from 수정

Related to : #116

jd99iam committed:
refactor: 부모가 없는 댓글들을 더하는 로직 수정

Related to : #116

jd99iam committed:
refactor: static 제거

Related to : #116

jd99iam committed:
refactor: toProblem 수정

Related to : #116

jd99iam committed:
Merge branch '1-develop' into refactor/116-project

jd99iam commented:
SolutionPostRequest의 toProblem 수정했습니다

jd99iam committed:
refactor: update 수정

1. 라인 정리
2. 변경감지 활용하도록 변경

Related to : #116

jd99iam committed:
refactor: delete 수정

1. 라인 정리

Related to : #116

jd99iam committed:
refactor: saveScrap 리팩터링

1. 라인 정리

Related to : #116

jd99iam committed:
refactor: getUpdateForm 리팩터링

1. 라인 정리

Related to : #116

jd99iam committed:
Merge remote-tracking branch 'upstream/refactor/116-project' into refactor/116-project

# Conflicts:
#	backend/src/main/java/com/prograngers/backend/service/SolutionService.java

jd99iam committed:
refactor: messages.properties 추가

리다이렉트 경로 message로 관리

Related to : #116

jd99iam committed:
refactor: messages.properties 추가

CommentController 리다이렉트 경로 message로 관리

Related to : #116

jd99iam committed:
refactor: LikesController 리다이렉트 경로 message로 관리

Related to : #116

jd99iam committed:
refactor: SolutionController 리다이렉트 경로 message로 관리

Related to : #116

jd99iam committed:
refactor: CommentController 메서드 추출

Related to : #116

jd99iam committed:
refactor: LikesController 메서드 추출

Related to : #116

jd99iam committed:
fix: 타인 프로필 보기 화면 풀이 개행문자 반영되도록 수정

Related to : #116

jd99iam committed:
fix: Comment 수정 후 반환하는 id 값 수정

Related to : #116

jd99iam committed:
fix: Comment 응답 dto에 isMine  추가

Related to : #116

jd99iam committed:
fix: SolutionReview에 내 리뷰인지 알 수 있도록 필드 추가

Related to : #116

jd99iam committed:
fix: SolutionReviewReply에 내 리뷰인지 알 수 있도록 필드 추가

Related to : #116

jd99iam committed:
fix: 상세보기 한 줄 리뷰에 @LogginedMember required false 추가

Related to : #116

jd99iam committed:
fix: 타인 프로필 보기 화면 pathvariable 닉네임으로 수정

Related to : #118
